pkg/plugins/plugin.go

```diff
@@ -0,0 +1,44 @@
+package plugins
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Plugin interface {
+	Init() error
+	Name() string
+	GetParam(string) string
+}
+
+// DecomposeCSL splits the comma separated list of plugins to be configured and registered
+func DecomposeCSL(in string) (map[string]Plugin, error) {
+	out := make(map[string]Plugin)
+	if in == "" {
+		return nil, fmt.Errorf("no plugins specified")
+	}
+
+	pStrList := strings.Split(in, ",")
+	if len(pStrList) == 0 {
```

Comment:
Should this really error if we enable plugins but there are none specified? IMO this is a valid scenario